synapse/app/homeserver.py

```diff
@@ -512,6 +522,17 @@ def generate_user_daily_visit_stats():
     # table will decrease
     clock.looping_call(generate_user_daily_visit_stats, 5 * 60 * 1000)
 
+    def generate_monthly_active_users():
+        count = 0
+        if hs.config.limit_usage_by_mau:
+            count = hs.get_datastore().count_monthly_users()
```

Comment:
`count_monthly_users` will need to be asynchronous and return a deferred, which we'll need to yield

Reply:
agreed as above

Reply:
This still needs a yield